torch/distributed/_functional_collectives.py

```diff
@@ -848,15 +885,18 @@ def mk_out_tensor(input):
 # but then you pass those sizes explicitly, and the all to all itself
 # isn't dynamic, it just follows the specified output splits
 def _all_to_all_single_meta(
-    input, output_split_sizes, input_split_sizes, *args, **kwargs
+    output_size: int,
+    input,
+    output_split_sizes: torch.Tensor,
+    input_split_sizes: torch.Tensor,
+    *args,
+    **kwargs,
```

Comment:
Chatted offline. The modification may not be necessary if the autograd function is called within the dispatcher op.